capnp/src/serialize.rs

```diff
@@ -86,6 +88,66 @@ pub fn read_message_from_flat_slice<'a>(slice: &mut &'a [u8],
     }
 }
 
+/// Segments read from a buffer, useful for long lived mmaps.
```

Comment:
My use case is that I own a message (as a `Vec<u8>`) and I need to return a string extracted from the message. Because of the `'a` associated to `self` in https://docs.capnproto-rust.org/capnp/message/struct.Reader.html#method.get_root I need to keep the `message::Reader` alive and store it *somewhere*.
However the only place I can store it is in the struct that already contains the message buffer, but then I cannot read it using the `SliceSegments` because it would make it a self-referential struct. I have two solutions:
* Read the buffer back with https://docs.capnproto-rust.org/capnp/serialize/fn.read_message.html at the cost of basically duplicating the message which I want to avoid for performance reasons
* Or use the API you introduced to actually *store* my buffer in the `BufferSegments` which is stored in the `Reader` without any copy or heap allocation :+1: 

All of that to say that this `BufferSegments` is not only useful for long lived mmaps but actually in many other scenario as well.

Reply:
I use it for that too, I have an enum with a memmap2 `Mmap` and a `Vec<u8>` and implement `Deref` on that. Grabbing large objects with `rusoto_s3` for these long lived messages is most naturally done by reading to `Vec<u8>`.